### PR TITLE
allow a custom config

### PR DESCRIPTION
**Pull request by digitalkaoz**
adds the ability to e.g. set custom formatters

### CODE REVIEW

File: sanic_json_logging/__init__.py [hunk 1]
```diff
@@ -27,13 +27,14 @@ def setup_json_logging(
...
     disable_json_access_log: bool = False,
+    config: dict = LOGGING_CONFIG_DEFAULTS
 ) -> None:
     """
     Sets up request logging
     """
     # Set up logging
-    LOGGING_CONFIG_DEFAULTS["formatters"]["generic"]["context"] = context_var
-    logging.config.dictConfig(LOGGING_CONFIG_DEFAULTS)
+    config["formatters"]["generic"]["context"] = context_var
```
Comment (terrycain) [inline sanic_json_logging/__init__.py:36]: If a user can override the value of `config`, they could omit generic and context dict's, can you add some error checking to add those in if they don't exist.

**Review by terrycain — CHANGES_REQUESTED** [review]

**Comment by terrycain** [comment]
Overall looks good :smile:

**Review by terrycain — APPROVED** [review]
I cant help think there's a nicer way do to it, but I cant think of any so am happy :D

**Comment by digitalkaoz** [comment]
@terrycain i wasnt perfectly happy either ;)

**Pull request merged**